ppb/assetlib.py

```diff
@@ -212,9 +212,13 @@ def load(self, timeout: float = None):
         Will block until the data is loaded.
         """
         # NOTE: This is called by FreeingMixin.__del__()
-        if not self.is_loaded() and not _executor.running():
-            logger.warning(f"Waited on {self!r} outside of the engine")
-        return self._future.result(timeout)
+        try:
+            return self._cached_result
```

Comment:
We should benchmark this independently. We used to do something like this in Vector and found just doing the work was cheaper than the cost of the look-ups.

Reply:
I definitely would want to benchmark this, but I will at least point out that in any case where we want to cache something doing it with a try/except block is cheaper than a lookup followed by a calculation because it incurs 0 costs on the hit branch. The try-except variant does not _have_ a cost of look-ups, necessarily.